Fix alignment on left border between menu navigation controls and menu item

jeryj opened the pull request:
Closes https://github.com/WordPress/gutenberg/issues/19510



## Description
I added 1px to the left alignment on the psuedo element that controls the left border on the menu item.

This was the best way I could see to fix this. Nothing else indicated a good option for adjusting spacing, and this should not affect the positioning of any other blocks or increase the CSS file size substantially (two added characters). 

I did not need to add any additional selectors or CSS (just changed a number that already existed in the CSS).

## How has this been tested?
Manually, via Chrome. I need to test additionally in more browsers.

## Screenshots
**Before**
<img width="258" alt="Left alignment off between block control and block" src="https://user-images.githubusercontent.com/967608/72001983-3184bb00-320c-11ea-9112-5952b9577ae0.png">

**After**
<img width="592" alt="Screen Shot 2020-01-08 at 11 20 44 AM" src="https://user-images.githubusercontent.com/967608/72002031-45c8b800-320c-11ea-9981-207f2e551c4d.png">

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->

obenland commented on packages/block-library/src/navigation/editor.scss:
```diff
@@ -42,7 +42,7 @@
 
 	.wp-block-navigation .block-editor-block-list__block::before {
-		left: 0;
+		left: 1px; // fixes a slight misalignment of the left border alignment between the menu item block control and the menu item border
```
Thanks for adding a comment with context! What caused that misalignment?

jeryj replied:
I can't quite figure that one out. I was trying to address this via a more top-level solution (addressing the positioning of the parent container) but didn't see anything off...

jeryj replied:
From a quick google search, I think most people who use it add it via a custom walker on the `wp_nav_menu()` php function.

obenland reviewed (changes requested):
I did some more debugging and found that not the block border needs adjustment, but the toolbar for navigation links

draganescu commented:
> I did some more debugging and found that not the block border needs adjustment, but the toolbar for navigation links

What did you find @obenland ? The toolbar has no changes as far as I can see in the navigation block, it is positioned based on the global $block-margin, which is the same everywhere. It appears that the blocks for navigation links have a weird margin/padding or something set.

jeryj commented:
@obenland, is the code @tellthemachines found what you're referencing? That does seem like a much better fix IMO. https://github.com/WordPress/gutenberg/issues/19510#issuecomment-572360093

obenland commented:
It is, should probably be changed to `$grid-size`, too. I was hoping it would give you a chance to debug for yourself and get to know the component a little better. Next time.

jeryj commented:
Don't worry, @obenland, I did debug it more and then saw @tellthemachines's comment later :)

jeryj committed:
Fix alignment of left border between navigation-link block and its block controls

Changed the hardcoded 15px value to the $block-padding variable (14px) to fix the one px offset between the lines.

jeryj commented:
@obenland, I ended up using`$block-padding`, as it was set to `14px` instead of `$grid-size`'s `8px`. I've removed the previous incorrect commit and pushed the new one.
